src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs

```diff
@@ -1405,5 +1407,58 @@ public void TerminationInNonBacktrackingVsBackTracking(RegexOptions options, int
             // NonBacktracking needs way less than 1s
             Assert.False(re.Match(input).Success);
         }
+
+        //
+        // dotnet/runtime-assets contains a set a regular expressions sourced from
+        // permissively-licensed packages.  Validate Regex behavior with those expressions.
+        //
+
+        [Theory]
+        [InlineData(RegexEngine.Interpreter)]
+        [InlineData(RegexEngine.Compiled)]
+        public void PatternsDataSet_ConstructRegexForAll(RegexEngine engine)
+        {
+            Parallel.ForEach(s_patternsDataSet.Value, r =>
+            {
+                try
+                {
+                    RegexHelpers.GetRegexAsync(engine, r.Pattern, r.Options).GetAwaiter().GetResult();
```

Comment:
is there anything useful we can do with the regex here? I guess I can't think of much unless you want to validate the ToString() == r.Pattern and Options = r.Options.

Reply:
I was thinking we might at least want to run IsMatch to at least force some portion of the code to execute, and just validate it successfully JITs / etc.  Beyond that, I'm not sure.  I expect subsequently we'll be looking at generating inputs for at least a subset of the expressions and then validating those inputs here.

Reply:
Yeah, I was talking offline with @danmoseley and it would be good if we add to that json some sample input that should match (perhaps also include the expected catpure groups too), and some that should not match. We then later can share this json with other engines as a test bed.